packages/wallets/src/features/cashier/modules/Transactions/components/TransactionsCompletedRow/components/TransactionsCompletedRowTransferAccountDetails/TransactionsCompletedRowTransferAccountDetails.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import { THooks, TWalletLandingCompanyName } from '../../../../../../../../types';
+import { getAccountName } from '../../../../../../helpers';
+import { TransactionsCompletedRowAccountDetails } from '../TransactionsCompletedRowAccountDetails';
+
+type TProps = {
+    accounts: THooks.AllAccountsList;
+    direction: 'from' | 'to';
+    loginid: string;
+};
+
+const TransactionsCompletedRowTransferAccountDetails: React.FC<TProps> = ({ accounts, direction, loginid }) => {
+    const wallet = accounts.wallets?.find(account => account.loginid === loginid);
+    if (wallet)
+        return (
+            <TransactionsCompletedRowAccountDetails
+                accountType={wallet.account_type ?? ''}
+                actionType='transfer'
+                currency={wallet.currency ?? 'USD'}
+                displayAccountName={getAccountName({
+                    accountCategory: 'wallet',
+                    //@ts-expect-error this needs backend typing
+                    accountType: wallet.account_type,
+                    displayCurrencyCode: wallet.currency_config?.display_code ?? 'USD',
+                    landingCompanyName: (wallet.landing_company_name ?? '') as TWalletLandingCompanyName,
+                })}
+                displayActionType={`Transfer ${direction}`}
+                isDemo={Boolean(wallet.is_virtual)}
+                isInterWallet
+            />
+        );
```

Comment:
```suggestion
    const wallet = accounts.wallets?.find(account => account.loginid === loginid);
    if (wallet)
        return (
            <TransactionsCompletedRowAccountDetails
                accountType={wallet.account_type ?? ''}
                actionType='transfer'
                currency={wallet.currency ?? 'USD'}
                displayAccountName={getAccountName({
                    accountCategory: 'wallet',
                    //@ts-expect-error this needs backend typing
                    accountType: wallet.account_type,
                    displayCurrencyCode: wallet.currency_config?.display_code ?? 'USD',
                    landingCompanyName: (wallet.landing_company_name ?? '') as TWalletLandingCompanyName,
                })}
                displayActionType={`Transfer ${direction}`}
                isDemo={Boolean(wallet.is_virtual)}
                isInterWallet
            />
        );
```
Judging from this, it seems that each of these are the same structure. I think u can do a single return for this if it is either (wallet/dtradeAccount/dxtradeAccount/mt5Account). The only difference is the `accountType` toggle

Reply:
Indeed, have done